File: Loenn/entities/boss_tiers.lua
local bossTiers = {}

-- Tier 1 Boss
local bossTier1 = {}
bossTier1.name = "Ingeste/BossTier1"
bossTier1.depth = 0
bossTier1.placements = {
    {
        name = "boss_tier_1",
        data = {
            bossType = "BasicEnemy",
            tier = 1,
            gimmick = 0,
            health = 100,
            speed = 20.0,
            arenaRadius = 150.0
        }
    }
}

-- Tier 2 Boss
local bossTier2 = {}
bossTier2.name = "Ingeste/BossTier2"
bossTier2.depth = 0
bossTier2.placements = {
    {
        name = "boss_tier_2",
        data = {
            bossType = "ElementalGuardian",
            tier = 2,
            gimmick = 1,
            health = 200,
            speed = 30.0,
            arenaRadius = 175.0
        }
    }
}

-- Tier 3 Boss
local bossTier3 = {}
bossTier3.name = "Ingeste/BossTier3"
bossTier3.depth = 0
bossTier3.placements = {
    {
        name = "boss_tier_3",
        data = {
            bossType = "ShadowWarrior",
            tier = 3,
            gimmick = 2,
            health = 350,
            speed = 45.0,
            arenaRadius = 200.0
        }
    }
}

-- Tier 4 Boss
local bossTier4 = {}
bossTier4.name = "Ingeste/BossTier4"
bossTier4.depth = 0
bossTier4.placements = {
    {
        name = "boss_tier_4",
        data = {
            bossType = "CrystalLord",
            tier = 4,
            gimmick = 4,
            health = 500,
            speed = 60.0,
            arenaRadius = 225.0
        }
    }
}

-- Tier 5 Boss
local bossTier5 = {}
bossTier5.name = "Ingeste/BossTier5"
bossTier5.depth = 0
bossTier5.placements = {
    {
        name = "boss_tier_5",
        data = {
            bossType = "VoidKnight",
            tier = 5,
            gimmick = 5,
            health = 750,
            speed = 80.0,
            arenaRadius = 250.0
        }
    }
}

-- Common field information for all tiers
local commonFieldInfo = {
    tier = {
        options = {
            ["Lowest"] = 1,
            ["Low"] = 2,
            ["Mid"] = 3,
            ["High"] = 4,
            ["Highest"] = 5,
            ["Final"] = 6
        }
    },
    gimmick = {
        options = {
            ["None"] = 0,
            ["Teleport"] = 1,
            ["Time Freeze"] = 2,
            ["Shield Breaker"] = 3,
            ["Elemental Fusion"] = 4,
            ["Gravity Control"] = 5,
            ["Dimension Rift"] = 6
        }
    }
}

bossTier1.fieldInformation = commonFieldInfo
bossTier2.fieldInformation = commonFieldInfo
bossTier3.fieldInformation = commonFieldInfo
bossTier4.fieldInformation = commonFieldInfo
bossTier5.fieldInformation = commonFieldInfo

bossTier1.texture = "objects/boss/tier1"
bossTier2.texture = "objects/boss/tier2"
bossTier3.texture = "objects/boss/tier3"
bossTier4.texture = "objects/boss/tier4"
bossTier5.texture = "objects/boss/tier5"

return {
    bossTier1,
    bossTier2,
    bossTier3,
    bossTier4,
    bossTier5
}

File: Source/Core/Entities/BossTiers.cs
namespace DesoloZantas.Core.Core.Entities
{
    // Tier 1 - Lowest Boss
    [CustomEntity("Ingeste/BossTier1")]
    [Tracked]
    public class BossTier1 : Boss
    {
        public BossTier1(EntityData data, Vector2 offset) : base(data, offset)
        {
            Tier = BossTier.Lowest;
            Gimmick = GimmickAbility.None;
            BossType = "BasicEnemy";
        }
    }

    // Tier 2 - Low Boss
    [CustomEntity("Ingeste/BossTier2")]
    [Tracked]
    public class BossTier2 : Boss
    {
        public BossTier2(EntityData data, Vector2 offset) : base(data, offset)
        {
            Tier = BossTier.Low;
            Gimmick = GimmickAbility.Teleport;
            BossType = "ElementalGuardian";
        }
    }

    // Tier 3 - Mid Boss
    [CustomEntity("Ingeste/BossTier3")]
    [Tracked]
    public class BossTier3 : Boss
    {
        public BossTier3(EntityData data, Vector2 offset) : base(data, offset)
        {
            Tier = BossTier.Mid;
            Gimmick = GimmickAbility.TimeFreeze;
            BossType = "ShadowWarrior";
        }
    }

    // Tier 4 - High Boss
    [CustomEntity("Ingeste/BossTier4")]
    [Tracked]
    public class BossTier4 : Boss
    {
        public BossTier4(EntityData data, Vector2 offset) : base(data, offset)
        {
            Tier = BossTier.High;
            Gimmick = GimmickAbility.ElementalFusion;
            BossType = "CrystalLord";
        }
    }

    // Tier 5 - Highest Boss
    [CustomEntity("Ingeste/BossTier5")]
    [Tracked]
    public class BossTier5 : Boss
    {
        public BossTier5(EntityData data, Vector2 offset) : base(data, offset)
        {
            Tier = BossTier.Highest;
            Gimmick = GimmickAbility.GravityControl;
            BossType = "VoidKnight";
        }
    }
}



